### PR TITLE
implement archiving attendees

### PR DESCRIPTION
**Pull request by moraleida**
### 🎫 Ticket

[ET-1322] 

### ✔️ Checklist
- [ ] I've included a changelog entry in the readme.txt file. 
- [ ] My code is tested. 
- [ ] My code has [proper inline documentation](https://the-events-calendar.github.io/products-engineering/docs/code-standards/).

[ET-1322]: https://theeventscalendar.atlassian.net/browse/ET-1322?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Tickets/Commerce/Flag_Actions/Archive_Attendees.php [hunk 1]
```diff
@@ -40,8 +40,9 @@ class Archive_Attendees extends Flag_Action_Abstract {
 	 * {@inheritDoc}
 	 */
 	public function handle( Status_Interface $new_status, $old_status, \WP_Post $post ) {
-		// @todo we need an error handling piece here.
-		if ( empty( $post->items ) ) {
+		$status_args = $new_status->get_wp_arguments();
+
+		if ( empty( $post->items ) || $status_args['label'] !== $post->status_name ) {
```
Comment (bordoni) [inline src/Tickets/Commerce/Flag_Actions/Archive_Attendees.php:45]: I dont follow this check.

File: src/Tickets/Commerce/Attendee.php [hunk 2]
```diff
@@ -228,6 +228,15 @@ public function register_post_type() {
...
 	}
 
+	/**
+	 * Archives an attendee. In WordPress this means the attendee post will have `trash` status, but it won't be deleted.
+	 *
+	 * @since TBD
+	 */
+	public function archive( $attendee_id ) {
+		wp_trash_post( $attendee_id );
+	}
```
Comment (bordoni) [inline src/Tickets/Commerce/Attendee.php:238]: Lets add Hooks.

File: src/Tickets/Commerce/Attendee.php [hunk 3]
```diff
@@ -339,15 +348,22 @@ public function create( \WP_Post $order, $ticket, array $args = [] ) {
...
 		$post = get_post( $post_id );
+		$flag = 'archive_attendees';
 
 		if ( static::POSTTYPE !== $post->post_type ) {
 			return;
 		}
 
+		if ( did_action( "tec_tickets_commerce_order_status_flag_{$flag}" ) ) {
+			return;
+		}
```
Comment (bordoni) [inline src/Tickets/Commerce/Attendee.php:365]: No need for the variable here.

**Review by bordoni — CHANGES_REQUESTED** [review]

File: src/Tickets/Commerce/Flag_Actions/Archive_Attendees.php (hunk 1 above)
Comment (moraleida) [reply src/Tickets/Commerce/Flag_Actions/Archive_Attendees.php:46]: This is in place to make sure that the status we're moving to in `$new_status` is the same as we already have in the order. So we don't make any changes to attendees if those two mismatch for some reason.

**Pull request merged**